cluster-autoscaler/cloudprovider/gce/mig_info_provider.go

```diff
@@ -158,12 +163,19 @@ func (c *cachingMigInfoProvider) findMigWithMatchingBasename(instanceRef GceRef)
 }
 
 func (c *cachingMigInfoProvider) fillMigInstances(migRef GceRef) error {
+	if val, ok := c.migInstancesLastRefreshedInfo[migRef.String()]; ok {
+		// do not regenerate MIG instances cache if last refresh happened in last migInstancesMaxRefreshRate.
+		if time.Now().Sub(val) < c.migInstancesMaxRefreshRate {
+			return nil
+		}
+	}
 	klog.V(4).Infof("Regenerating MIG instances cache for %s", migRef.String())
 	instances, err := c.gceClient.FetchMigInstances(migRef)
 	if err != nil {
 		c.migLister.HandleMigIssue(migRef, err)
 		return err
 	}
+	c.migInstancesLastRefreshedInfo[migRef.String()] = time.Now()
```

Comment:
this only sets the refresh value for a successful fetch which return instances.
Don't we also want to wait for the same timeout for unsuccessful fetches?

Maybe we don't, can you drop a small rationale for it tho.

Reply:
added a comment, I don't think we should cache unsuccessful results, as the issues can be transient with those so another call might make sense